File: C++/funciones.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    // Arrays de números
    int a[] = {1, 2, 3, 4, 5};
    int b[] = {6, 7, 8, 9, 10};
    vector<int> c;

    // Multiplicar los valores correspondientes y almacenarlos en 'c'
    for (int i = 0; i < 5; i++) {
        c.push_back(a[i] * b[i]);
    }

    // Imprimir los resultados
    cout << "Resultado de la multiplicación: ";
    for (int i = 0; i < c.size(); i++) {
        cout << c[i] << " ";
    }
    cout << endl;

    return 0;
}

// Definición de funciones

// Función para mostrar un texto
void mostrar_texto() {
    cout << "Hola" << endl;
}

// Función para multiplicar dos números fijos
void multiplicar() {
    int a = 5, b = 8;
    cout << a * b << endl;
}

// Multiplicación con variables globales
int a = 5, b = 8;
void multiplicar_global() {
    cout << a * b << endl;
}

// Función que devuelve un resultado
int multiplicar_con_return() {
    int a = 5, b = 8;
    return a * b;
}

// Función que devuelve un booleano según una condición
bool validar_dato() {
    if (a == 5) {
        return true;
    } else {
        return false;
    }
}

int main_functions() {
    // Llamadas a las funciones
    mostrar_texto();
    multiplicar();
    multiplicar_global();

    int Resultado = multiplicar_con_return();
    cout << "Resultado + 5: " << Resultado + 5 << endl;

    bool dato = validar_dato();
    cout << "Validación del dato: " << dato << endl;

    return 0;
}

File: JAVA/funciones.java
import java.util.ArrayList;
import java.util.List;

public class Funciones {
    public static void main(String[] args) {
        // Definir las listas (tuplas en Python equivalen a arrays o listas en Java)
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {6, 7, 8, 9, 10};
        List<Integer> c = new ArrayList<>();

        // Multiplicar elementos y agregar a la lista C
        for (int i = 0; i < a.length; i++) {
            c.add(a[i] * b[i]);
        }

        // Imprimir el resultado
        System.out.println(c);

        // Llamadas a funciones
        mostrarTexto();
        multiplicar();
        
        // Ejemplo de variables globales
        int resultado = multiplicarConRetorno();
        System.out.println(resultado + 5);

        // Validación de dato
        int datoA = 5;
        boolean esValido = validarDato(datoA);
        System.out.println(esValido);
    }

    // Función sin parámetros ni retorno
    public static void mostrarTexto() {
        System.out.println("Hola");
    }

    // Función sin parámetros ni retorno que realiza una multiplicación
    public static void multiplicar() {
        int a = 5;
        int b = 8;
        System.out.println(a * b);
    }

    // Función que devuelve un resultado
    public static int multiplicarConRetorno() {
        int a = 5;
        int b = 8;
        return a * b;
    }

    // Función que valida un dato y devuelve un booleano
    public static boolean validarDato(int a) {
        return a == 5;
    }
}
